# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/valentin/kismet/gcc34.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ruN kismet-2004-04-R1/packet.h kismet-2004-04-R1-new/packet.h
--- kismet-2004-04-R1/packet.h	2004-04-02 18:53:36.000000000 +0200
+++ kismet-2004-04-R1-new/packet.h	2004-05-24 13:58:03.682859328 +0200
@@ -550,7 +550,7 @@
     // This isn't quite like STL iterators, because I'm too damned lazy to deal with all
     // the nasty STL hoop-jumping.  This does provide a somewhat-stl-ish interface to
     // iterating through the singleton and masked maps
-    friend class iterator {
+    class iterator {
         friend class macmap;
 
     public:
@@ -643,6 +643,7 @@
         int vector_itr;
         macmap<T> *owner;
     };
+    friend class iterator;
 
     iterator begin() {
         iterator ret(this);
